modules/features2d/src/orb.cpp

```diff
@@ -1109,9 +1129,10 @@ void ORB_Impl::detectAndCompute( InputArray _image, InputArray _mask,
             nkeypoints = (int)keypoints.size();
             for( i = 0; i < nkeypoints; i++ )
             {
-                level = keypoints[i].octave;
-                CV_Assert(0 <= level);
-                allKeypoints[level].push_back(keypoints[i]);
+                level = getLevel(keypoints[i]);
+                keypoints[i].octave = level + firstLevel;
+                CV_Assert(keypoints[i].octave >= 0);
```

Comment:
@vpisarev Finally, all negative values are corrected using the `firstLevel` offset, and the assertion here ensures that the final value is just OK